test-workloads/src/bin/tellus.rs

```diff
@@ -344,6 +374,33 @@ extern "C" fn kernel_init(hart_id: u64, fdt_addr: u64) {
                         }
                         zero_pages_added += 1;
                     }
+                    sbi::TvmCpuExitCode::SharedPageFault => {
+                        // Figure out where the fault occurred.
+                        let fault_addr = get_fault_address(vmid);
+                        if fault_addr != SHARED_PAGES_START_ADDRESS {
+                            println!("Fault in shared page region at {fault_addr:x}");
+                            panic!("Unexpected shared page fault address");
+                        }
+                        let msg = SbiMessage::Tee(sbi::TeeFunction::TvmAddSharedPages {
+                            guest_id: vmid,
+                            page_addr: next_page,
```

Comment:
I'd save the address of the shared page in a separate variable in case someone else inserts code in between that bumps `next_page`.